tests/Adapters/RackspaceConnectorTest.php

```diff
@@ -122,6 +122,23 @@ public function testConnectWithoutContainer()
         ]);
     }
 
+    /**
+     * @expectedException \Guzzle\Http\Exception\ClientErrorResponseException
+     */
+    public function testConnectWithInternal()
+    {
+        $connector = $this->getRackspaceConnector();
+
+        $connector->connect([
+            'endpoint'  => 'https://lon.identity.api.rackspacecloud.com/v2.0/',
+            'region'    => 'LON',
+            'username'  => 'your-username',
+            'apiKey'    => 'your-api-key',
+            'container' => 'your-container',
+            'internal'  => true,
```

Comment:
And what if internal is false. We need more test cases. We also need test cases that result in your new logic throwing an exception.